glad/parse.py

```diff
@@ -1453,3 +1478,52 @@ def from_element(cls, element, api=None):
     def __str__(self):
         return '{self.name}@{self.version!r}'.format(self=self)
     __repr__ = __str__
+
+
+class SpecificationDocs(object):
+    URL = None
+
+    def __init__(self, api, version, profile, extensions):
+        self.api = api
+        self.version = version
+        self.profile = profile
+        self.extensions = extensions
+
+    def load(self):
+        raise NotImplementedError
+
+    def docs_for_command_name(self, name):
+        """
+        Returns the CommandDocs for the given command name or None if not found.
+        """
+        raise NotImplementedError
+
+
+class CommandDocs(object):
+    """
+    Inline code documentation for a command/function.
+    """
+
+    # Template rendering will interpret this as a custom paragraph break.
+    # If no special break logic is needed, just use '\n'.
+    BREAK = '__BREAK__'
```

Comment:
I think it's better if instead we just had a list of paragraphs instead of joining and then splitting it after again.

Reply:
Paragraphs are list of strings joined by a like break. However I needed another level of "breaking" logic, for example, for list items, term definitions, etc. I'll try to remove this logic.

Reply:
I am not too worried about, I was just thinking instead of keeping the breaks in the text, we can just keep a list of text instead.